.github/workflows/discord-posts.yml

```diff
@@ -39,7 +39,14 @@ jobs:
           PR_NUMBER: ${{ github.event.pull_request.number }}
           PR_URL: ${{ github.event.pull_request.html_url }}
           CHECK_RESULT: ${{ env.CHECK_RESULT }}
+          PING_ON_FAILURE: ${{ secrets.DEV_OPS_ROLE_ID }}
         run: |
-          curl -X POST -H 'Content-Type: application/json' -d '{
-            "content": "'"PR merged: [(#${PR_NUMBER}) ${PR_TITLE}](${PR_URL})\\n${CHECK_NAME} result: ${CHECK_RESULT}"'"
-          }' ${DISCORD_WEBHOOK_URL}
```

Comment:
The message content was getting long and complex enough that I refactored it.

Technically, most of this PR is the refactor, and only the `PING_ON_FAILURE` above and the `if` block below are the functional changes.